File: src/main/java/com/tartner/dancehours/web/login/CreateAccountFormValidator.java
package com.tartner.dancehours.web.login;


import org.springframework.stereotype.Component;
import org.springframework.validation.Errors;
import org.springframework.validation.Validator;

@Component
public class CreateAccountFormValidator implements Validator {
    @Override
    public boolean supports( final Class<?> clazz ) {
        return CreateAccountForm.class.isAssignableFrom( clazz );
    }
    @Override
    public void validate( final Object target, final Errors errors ) {
        CreateAccountForm form = (CreateAccountForm) target;
        // TODO: custom validation?

    }
}

File: src/main/java/com/tartner/dancehours/web/login/CreateAccountFormValidator.kt
package com.tartner.dancehours.web.login


import org.springframework.stereotype.Component
import org.springframework.validation.Errors
import org.springframework.validation.Validator

@Component open class CreateAccountFormValidator : Validator {
    override fun supports(clazz: Class<*>): Boolean {
        return CreateAccountForm::class.java.isAssignableFrom(clazz)
    }

    override fun validate(target: Any, errors: Errors) {
        val form = target as CreateAccountForm
        // TODO: custom validation?

    }
}
